src/inmanta/protocol/openapi/converter.py

```diff
@@ -180,18 +186,24 @@ def _handle_list(self, type_annotation: Type) -> Schema:
         return Schema(type="array", items=self.get_openapi_type(list_member_type[0]))
 
     def get_openapi_type(self, type_annotation: Type) -> Schema:
-
-        if typing_inspect.is_optional_type(type_annotation):
+        if (
+            typing_inspect.is_optional_type(type_annotation)
+            and len(typing_inspect.get_args(type_annotation, evaluate=True)) == 2
+        ):
             return self._handle_optional_type(type_annotation)
```

Comment:
Please remove and handle case this in _handle_union_type. The current _handle_optional_type implementation looks a bit fishy and can not, in fact, correctly handle all optional types, as opposed to  _handle_union_type.

Reply:
(also add a comment to union that an optional is a union)